<launch>
  <!-- number of moving objects -->
  <arg name="obj_num" value="10" />
  <arg name="drone_id" value="1"/>

  <arg name="map_size_x" value="26.0"/>
  <arg name="map_size_y" value="25.0"/>
  <arg name="map_size_z" value=" 4.0"/>
  <arg name="init_x" value="  0.0"/>
  <arg name="init_y" value="  0.0"/>
  <arg name="init_z" value="  0.0"/>
  <arg name="number_sample" value="500"/>
  <arg name="odom_topic" value="visual_slam/odom" />
  
  <!-- map -->
  <node pkg="mockamap" type="mockamap_node" name="mockamap_node" output="screen">
    <param name="seed" type="int" value="19"/>
    <param name="update_freq" type="double" value="1.0"/>

    <!--  box edge length, unit meter-->
    <param name="resolution" type="double" value="0.18"/>

    <!-- map size unit meter-->
    <param name="x_length" type="int" value="26"/>
    <param name="y_length" type="int" value="25"/>
    <param name="z_length" type="int" value="4"/>

    <!-- <param name="type" type="int" value="4"/>
    <param name="numNodes"        type="int"    value="4"/>
    <param name="connectivity"        type="double" value="1.0"/>
    <param name="roadRad"        type="int" value="12"/>
    <param name="nodeRad"        type="int" value="12"/> -->

    <!-- <param name="type" type="int" value="3"/>
    <param name="road_width"        type="double"    value="1"/>
    <param name="add_wall_x"        type="int" value="0"/>
    <param name="add_wall_y"        type="int" value="0"/>
    <param name="maze_type"        type="int" value="1"/> -->

    <param name="type" type="int" value="2"/>
    <param name="width_min"        type="double"    value="1"/>
    <param name="width_max"        type="double" value="2"/>
    <param name="obstacle_number"        type="int" value="30"/>

    <!-- <param name="type" type="int" value="1"/>
    <param name="complexity"        type="double"    value="0.142857"/>
    <param name="fill"        type="double" value="0.38"/>
    <param name="fractal"        type="int" value="1"/>
    <param name="attenuation"        type="double" value="0.5"/> -->
    
    <!--    maze type: -->
  </node>
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find so3_quadrotor_simulator)/launch/default4.rviz" required="true" />

  <node name="rrt_planner" pkg="rrt_planner" type="rrt_planner_node" output="screen" >
      <param name="number_sample" value="$(arg number_sample)" />
      <param name="mode" value="1" />
      <param name="map_x_size" value="$(arg map_size_x)" />
      <param name="map_y_size" value="$(arg map_size_y)" />
      <param name="map_z_size" value="$(arg map_size_z)" />
      <remap from="/grid_map/x_size" to="/mockamap_node/x_length" />
      <remap from="/grid_map/y_size" to="/mockamap_node/y_length" />
      <remap from="/grid_map/z_size" to="/mockamap_node/z_length" />
      <remap from="~cloud_in" to="/map_generator/global_cloud" />
      <param name="resolution" value="0.05" />
      <param name="obstacle_inflation" value="0.2" />
  </node>
  <node name="rrt_star_planner" pkg="rrt_planner" type="rrt_planner_node" output="screen" >
      <param name="number_sample" value="$(arg number_sample)" />
      <param name="mode" value="2" />
      <param name="map_x_size" value="$(arg map_size_x)" />
      <param name="map_y_size" value="$(arg map_size_y)" />
      <param name="map_z_size" value="$(arg map_size_z)" />
      <remap from="/grid_map/x_size" to="/mockamap_node/x_length" />
      <remap from="/grid_map/y_size" to="/mockamap_node/y_length" />
      <remap from="/grid_map/z_size" to="/mockamap_node/z_length" />
      <remap from="~cloud_in" to="/map_generator/global_cloud" />
      <param name="resolution" value="0.05" />
      <param name="obstacle_inflation" value="0.2" />
  </node>
  <node name="informed_rrt_star_planner" pkg="rrt_planner" type="rrt_planner_node" output="screen" >
      <param name="number_sample" value="$(arg number_sample)" />
      <param name="mode" value="3" />
      <param name="map_x_size" value="$(arg map_size_x)" />
      <param name="map_y_size" value="$(arg map_size_y)" />
      <param name="map_z_size" value="$(arg map_size_z)" />
      <remap from="/grid_map/x_size" to="/mockamap_node/x_length" />
      <remap from="/grid_map/y_size" to="/mockamap_node/y_length" />
      <remap from="/grid_map/z_size" to="/mockamap_node/z_length" />
      <remap from="~cloud_in" to="/map_generator/global_cloud" />
      <param name="resolution" value="0.05" />
      <param name="obstacle_inflation" value="0.2" />
  </node>
  <node name="prm_planner" pkg="planner" type="planner_node" output="screen" >
      <param name="number_sample" value="$(arg number_sample)" />
      <param name="map_x_size" value="$(arg map_size_x)" />
      <param name="map_y_size" value="$(arg map_size_y)" />
      <param name="map_z_size" value="$(arg map_size_z)" />
      <remap from="/grid_map/x_size" to="/mockamap_node/x_length" />
      <remap from="/grid_map/y_size" to="/mockamap_node/y_length" />
      <remap from="/grid_map/z_size" to="/mockamap_node/z_length" />
      <remap from="~cloud_in" to="/map_generator/global_cloud" />
      <param name="resolution" value="0.05" />
      <param name="obstacle_inflation" value="0.2" />
  </node>

</launch>
